src/time-to-interactive/TTIBoomerang.ts

```diff
@@ -0,0 +1,331 @@
+                    totalIntervals * this.CHECK_PERIOD >
+                    this.TTI_RESOLVE_TIMEOUT
+                ) {
+                    // TTI did not resolve in timeout period, so TTI can't be computed
+                    this.ttiResolved = true;
+                    this.ttiTracker = {};
+                    reject('TTI computation timed out');
+                }
+                const endBucket = this.computeTimeWindow();
+                for (let bucket = currBucket; bucket <= endBucket; bucket++) {
+                    currBucket = bucket;
+                    let allOk = true;
+
+                    if (!this.ttiTracker[LONG_TASK] && !this.ttiTracker[FPS]) {
+                        // can't resolve tti so wait and try again
+                        break;
+                    }
+
+                    // Check long task
+                    if (
+                        this.ttiTracker[LONG_TASK] &&
+                        this.ttiTracker[LONG_TASK][bucket] &&
+                        this.ttiTracker[LONG_TASK][bucket] >
+                            this.LONG_TASK_THRESHOLD
+                    ) {
+                        allOk = false;
+                    }
+
+                    // Check FPS
+                    if (
+                        this.fpsSupported &&
+                        this.ttiTracker[FPS] &&
+                        this.ttiTracker[FPS][bucket] &&
+                        this.ttiTracker[FPS][bucket] < this.FPS_THRESHOLD
+                    ) {
+                        allOk = false;
+                    }
+
+                    if (allOk) {
+                        okIntervals += 1;
+                        if (okIntervals >= this.REQUIRED_OK_INTERVALS) {
+                            this.ttiResolved = true;
+                            clearInterval(tti);
+                            // TTI is Visually Ready + time to start of ok period from visually ready
+                            const timeToQuietPeriodFromVisuallyReady: number =
+                                (currBucket -
+                                    this.REQUIRED_OK_INTERVALS -
+                                    startBucket +
+                                    1) *
+                                this.COLLECTION_PERIOD;
+                            resolve(
+                                visuallyReadyTimestamp +
+                                    timeToQuietPeriodFromVisuallyReady
+                            );
+                            break;
+                        } else {
+                            continue;
+                        }
+                    } else {
+                        okIntervals = 0;
+                    }
+
+                    currBucket = endBucket;
+                }
+                totalIntervals += 1;
+            }, this.CHECK_PERIOD);
+        });
+    }
+
+    initListeners(): void {
+        // record domcontentloaded if its found already
+        if (window.performance) {
```

Comment:
Question: how are you capturing `domContentLoadedEventEnd` if it fires after this check?